Exiled.Events/Patches/Events/Player/Handcuffing.cs

```diff
@@ -7,101 +7,78 @@
-                        {
-                            flag = true;
-                        }
-
-                        break;
-                }
-
-                if (!flag)
-                    return false;
-
-                __instance.ClearTarget();
-
-                var ev = new HandcuffingEventArgs(API.Features.Player.Get(__instance.gameObject), API.Features.Player.Get(target));
-
-                Player.OnHandcuffing(ev);
-
-                if (!ev.IsAllowed)
-                    return false;
-
-                handcuffs.NetworkCufferId = __instance.MyReferenceHub.queryProcessor.PlayerId;
-
-                return false;
-            }
-            catch (Exception e)
+            var newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
+
+            // The index offset.
+            var offset = 0;
+
+            // Search for the last "ldloc.2".
+            var index = newInstructions.FindLastIndex(instruction => instruction.opcode == OpCodes.Ldloc_2) + offset;
+
+            // Get the first label of the last "ret".
+            var returnLabel = newInstructions[newInstructions.Count - 1].labels[0];
+
+            // var ev = new HandcuffingEventArgs(Player.Get(this.gameObject), Player.Get(target), true);
+            //
+            // Handlers.Player.OnHandcuffing(ev);
+            //
+            // if (!ev.IsAllowed)
+            //   return;
+            newInstructions.InsertRange(index, new[]
             {
-                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.Handcuffing: {e}\n{e.StackTrace}");
+                // Player.Get(this.gameObject)
+                new CodeInstruction(OpCodes.Ldarg_0),
+                new CodeInstruction(OpCodes.Call, PropertyGetter(typeof(Component), nameof(Component.gameObject))),
+                new CodeInstruction(OpCodes.Call, Method(typeof(Player), nameof(Player.Get), new[] { typeof(GameObject) })),
+
+                // Player.Get(target)
+                new CodeInstruction(OpCodes.Ldarg_1),
+                new CodeInstruction(OpCodes.Call, Method(typeof(Player), nameof(Player.Get), new[] { typeof(GameObject) })),
+
+                // true
+                new CodeInstruction(OpCodes.Ldc_I4_1),
+
+                // var ev = new HandcuffingEventArgs(...);
+                new CodeInstruction(OpCodes.Newobj, GetDeclaredConstructors(typeof(HandcuffingEventArgs))[0]),
+                new CodeInstruction(OpCodes.Dup),
+
+                // Handlers.Player.OnHandcuffing(ev);
+                new CodeInstruction(OpCodes.Call, Method(typeof(Handlers.Player), nameof(Handlers.Player.OnHandcuffing))),
+
+                // if (!ev.IsAllowed)
+                //   return;
+                new CodeInstruction(OpCodes.Callvirt, PropertyGetter(typeof(HandcuffingEventArgs), nameof(HandcuffingEventArgs.IsAllowed))),
+                new CodeInstruction(OpCodes.Brfalse_S, returnLabel),
```

Comment:
Use `brfalse` instead

Reply:
Since it's a long method, it could be useful to not break it if they add more instructions.